### DISCUSSION TITLE
Syncing sessions from Garmin Connect

### DISCUSSION THREAD

**Discussion by gamestic** [djong1/gp3s#230] [General]
Hi, before I raise an issue, I wanted to check if I am doing something wrong.

I used to have a Suunto watch, and when I saved a windsurf session it synced with the Suunto app and was immediately available to post on the website. That watch unfortunately fell to the depths of the sea and is no more.

So I got a new watch, a Garmin Instinct 2X. Now, when I do a windsurf session and save it, it syncs automatically to Garmin Connect, as usual. I have connected my account with the Garmin Connect api under the My Devices section, but there is never a session to post.

In order to post a session I end up logging in to Garmin Connect, finding my activity, downloading the .fit file and then uploading that to the website and then it is available to post.

What am I doing wrong? I can share the activity on Garmin Connect if it helps. They are by default private, I don't think that should be the issue, as they were private on Suunto as well. Or does the Garmin Connect integration just not work at present?

Thanks,
James.

**Comment by jhdebruin** [comment]
Hi,

Please check out our Wiki page here -> https://github.com/djong1/gp3s/wiki/How-to-configure-your-Garmin-Device

You can also send the track file to us at info@gps-speedsurfing.com

**Comment by gamestic** [comment]
Hi, thanks for your reply. I checked the wiki page, I had already set the data recording to 1s interval. I am not sure what the .gpx requirement means, since I uploaded the .fit file successfully and posted that. I had already connected my device with Garmin Connect, so I think that all is ok.

I checked the app linked in the wiki, it is a data field and is not compatible with my watch. I am not sure what extra it would do for me?

Given that I can pull the original .fit file directly from Garmin Connect and upload that to the website and then post it successfully, I am not sure why this isn't happening automatically via the API as it used to with my old Suunto watch?

This is the session I posted via the .fit file:
https://www.gps-speedsurfing.com/default.aspx?mnu=user&val=427160&uid=25340&spotid=19540

I will email you the .fit file if required?

Thanks!

**Comment by jhdebruin** [comment]
The Garmin should be working as described, we see no additional errors on our side.  Could you please re-establish the connection with Garmin Connect and check if its working like designed. Please notice only session that are registered after the connection has been made will be synced.  

As far as we know the Instinct series doesn't support custom apps , so you can ignore my previous comment :)

**Comment by gamestic** [comment]
I have made the connection, actually a few times already... Here's a screenshot from Garmin Connect:

![Screenshot 2023-11-16 15 54 25](https://github.com/djong1/gp3s/assets/73649075/27a80fc2-bf58-4f6c-afa8-1922d45053f5)

This was the activity on Garmin Connect:

![Screenshot 2023-11-16 15 56 12](https://github.com/djong1/gp3s/assets/73649075/f969b407-71bb-40c0-93c3-64f2f045dd32)

I've made it public now:

https://connect.garmin.com/modern/activity/12741791874

I am going to windsurf again at the weekend, so I will grab more screenshots then and see what happens. And will update you soon.

Thanks!

**Comment by gamestic** [comment]
van broke down, couldn't windsurf :( will update again next time i go.

**Comment by gamestic** [comment]
This is all working correctly for me now, so closing.
thanks.
